packages/core-interfaces/src/core-container/container.ts

```diff
@@ -5,14 +5,18 @@ export interface PluginDescriptor {
     pkg: any;
     defaults?: any;
     extends?: string;
-    register(container: IContainer, options?: any): Promise<any>;
+    register(container: IContainer, options?: PluginRegisterOptions): Promise<any>;
     deregister?(container: IContainer, options?: any): Promise<void>;
 }
 
+export interface PluginRegisterOptions {
```

Comment:
Interfaces should always start with `I` - https://palantir.github.io/tslint/rules/interface-name and `IPluginOptions` might be a better name as the options are for the plugin overall _(used after the registration)_ not just the registration process.